pkg/segment/integrations.go

```diff
@@ -27,6 +27,10 @@ func getTelemetryForDevfileRegistry() (registryLibrary.TelemetryData, error) {
 		return td, nil
 	}
 
+	if GetDebugTelemetryFile() != "" {
+		return td, nil
+	}
+
```

Comment:
To make this faster, perhaps you can put this check on L19, before checking if telemetry is enabled.